core/chains/constraints.go

```diff
@@ -0,0 +1,18 @@
+package chains
+
+// ID types represent unique identifiers within a particular chain type. Using string is recommended.
+type ID any
```

Comment:
IIUC it is not possible to define a constraint that would capture (`string` | `Stringer`), but we may be able to switch over to just using `string` later on, at least throughout the lower levels where we often stringify the `utils.Big`s to use as map keys anyways.